inc/debug.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#if defined(CH_CFG_TIME_QUANTUM)
+    // We are compiling for ChibiOS
+    #include "osal.h"
+    #define debug_assert(condition, remark) osalDbgAssert(condition, remark)
+    #define debug_check(condition) osalDbgCheck(condition)
+#else
+    // We are not compiling for ChibiOS
+    #include <cassert>
+    #define debug_assert(condition, remark) assert(condition)
+    #define debug_check(condition) assert(condition)
+#endif
```

Comment:
i generally like to do:
```cpp
#endif // defined(CH_CFG_TIME_QUANTUM) 
```
as it makes it easier to track if multiple compiler directives are used